website/next.config.js

```diff
@@ -23,56 +21,85 @@ const env = {
 }
 
 const withNextra = nextra({
-  theme: './src/nextra-theme/index.tsx',
+  theme: './src/NextraLayout.tsx',
   search: false,
   codeHighlight: false,
   defaultShowCopyCode: false,
   readingTime: true,
   transformPageMap(pageMap) {
-    const locale = pageMap[0].data.slice(0, 2)
+    const locale = pageMap[0].route.slice(1, 3)
+    // TODO: remove this when crowdin will translate global.json
+    const t = (key) => {
+      try {
+        return translate(translations, locale, `global.sidebar.${key}`)
+      } catch {
+        return translate(translations, 'en', `global.sidebar.${key}`)
+      }
+    }
+
+    const metaFile = {
+      index: t('index'),
+      '---1': {
+        type: 'separator',
+      },
+      about: '',
+      network: t('network'),
+      sunrise: '',
+      billing: '',
+      glossary: '',
+      tokenomics: '',
+      arbitrum: t('arbitrum'),
+      '---2': {
+        type: 'separator',
+      },
+      '###1': {
+        type: 'heading',
+        title: t('subgraphs'),
+      },
+      'quick-start': '',
+      developing: t('developing'),
+      deploying: t('deploying'),
+      publishing: t('publishing'),
+      managing: t('managing'),
+      querying: t('querying'),
+      cookbook: t('cookbook'),
+      'release-notes': t('releaseNotes'),
+      '---3': {
+        type: 'separator',
+      },
+      '###2': {
+        type: 'heading',
+        title: t('substreams'),
+      },
+      substreams: '',
+      '---4': {
+        type: 'separator',
+      },
+      '###3': {
+        type: 'heading',
+        title: t('indexing'),
+      },
+      'operating-graph-node': '',
+      'chain-integration-overview': '',
+      'supported-network-requirements': '',
+      'new-chain-integration': '',
+      firehose: '',
+      graphcast: '',
+      'mips-faqs': '',
+    }
+
     return [
-      ...pageMap,
+      { data: metaFile },
       {
         route: `/${locale}`,
         name: 'index',
-        frontMatter: {
-          title: translate(translations, locale, 'index.title'),
-        },
+        frontMatter: {},
```

Comment:
I'm even more confused about this now because didn't you say [here](https://github.com/graphprotocol/docs/pull/504#discussion_r1696702807) that this page map entry was needed to translate the title, and now you remove the translation?

Reply:
I removed it here because I added in `meta` object https://github.com/graphprotocol/docs/blob/f5f1cb0382894a41c5436928b0e3b049bfc9320e/website/next.config.js#L40-L41

there is no need to duplication with this change

Reply:
Ah got it. But what happens if this page map entry is removed? The index page will 404?

Reply:
I guess it is to pass this check https://github.com/graphprotocol/docs/blob/5634476928786ea9aba552dd17c19574885ebee8/packages/nextra-theme/src/index.tsx#L104-L111